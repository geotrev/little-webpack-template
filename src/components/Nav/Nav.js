import React, { Component } from 'react';
import ReactDOM from 'react-dom';
import './Nav.scss';

import { NavLink } from 'react-router-dom';

export default class Nav extends Component {
  render() {
    return (
      <nav className="nav-wrapper">
        <ul>
          <li>
            <NavLink activeClassName="selected" exact to='/'>Home</NavLink>
          </li>
          <li>
            <NavLink activeClassName="selected" exact to='/about'>About</NavLink>
          </li>
        </ul>
      </nav>
    );
  }
}
