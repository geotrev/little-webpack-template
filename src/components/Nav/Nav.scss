.nav-wrapper {
  display: flex;
  flex: 0 0 100%;
  max-width: 100%;
  font-family: sans-serif;
  justify-content: center;

  ul {
    display: flex;
    margin: 0;
    padding: 0;
    list-style-type: none;
  }

  li a,
  li a:visited,
  li a:focus,
  li a:hover,
  li a:active {
    display: flex;
    color: gray;
    text-decoration: none;
    padding: 1rem;

    &.selected {
      border-bottom: 2px solid black;
      color: black;
    }
  }
}
