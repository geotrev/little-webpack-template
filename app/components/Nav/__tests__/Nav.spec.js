import React from "react"
import Nav from "../Nav"
import { BrowserRouter as Router } from "react-router-dom"

describe("<Nav />", () => {
  it("renders", () => {
    const wrapper = mount(
      <Router>
        <Nav />
      </Router>,
    )
    expect(wrapper).toMatchSnapshot()
  })
})
