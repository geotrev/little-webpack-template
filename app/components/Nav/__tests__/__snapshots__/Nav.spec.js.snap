// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Nav /> renders 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Nav>
      <nav
        className="nav-wrapper"
      >
        <ul>
          <li>
            <NavLink
              activeClassName="selected"
              exact={true}
              to="/"
            >
              <Link
                aria-current="page"
                className="selected"
                style={Object {}}
                to={
                  Object {
                    "hash": "",
                    "pathname": "/",
                    "search": "",
                    "state": null,
                  }
                }
              >
                <LinkAnchor
                  aria-current="page"
                  className="selected"
                  href="/"
                  navigate={[Function]}
                  style={Object {}}
                >
                  <a
                    aria-current="page"
                    className="selected"
                    href="/"
                    onClick={[Function]}
                    style={Object {}}
                  >
                    Home
                  </a>
                </LinkAnchor>
              </Link>
            </NavLink>
          </li>
          <li>
            <NavLink
              activeClassName="selected"
              exact={true}
              to="/about"
            >
              <Link
                aria-current={null}
                to={
                  Object {
                    "hash": "",
                    "pathname": "/about",
                    "search": "",
                    "state": null,
                  }
                }
              >
                <LinkAnchor
                  aria-current={null}
                  href="/about"
                  navigate={[Function]}
                >
                  <a
                    aria-current={null}
                    href="/about"
                    onClick={[Function]}
                  >
                    About
                  </a>
                </LinkAnchor>
              </Link>
            </NavLink>
          </li>
        </ul>
      </nav>
    </Nav>
  </Router>
</BrowserRouter>
`;
